dev-lang/icon/icon-9.5.20i.ebuild

```diff
@@ -0,0 +1,124 @@
+IUSE="X iplsrc"
+
+S="${WORKDIR}/icon-${PV}"
+
+RDEPEND="
+	X? ( x11-libs/libX11:= )"
+DEPEND="
+	${RDEPEND}
+	X? (
+		x11-libs/libXpm
+		x11-libs/libXt
+	)"
+
+src_prepare() {
+	default
+	# Add CFLAGS and LDFLAGS to Makefiles as needed
+	sed -i -e 's/CFLAGS="$(CFLAGS)"/& LDFLAGS="$(LDFLAGS)"/g' ipl/cfuncs/Makefile || die
+	sed -i -e 's/^CFLAGS = \(.*\)/CFLAGS = $(CFLAGS) \1/g' ipl/packs/loadfunc/Makefile || die
+	sed -i -e 's/CFLAGS="$(CFLAGS)"/& LDFLAGS="$(LDFLAGS)"/g' ipl/packs/loadfunc/Makefile || die
+	sed -i -e 's/-o $LIBNAME $CFLAGS/& $LDFLAGS/g' ipl/cfuncs/mklib.sh || die
+	sed -i -e 's/$LIBNAME -fPIC/$LIBNAME -fPIC $LDFLAGS/g' ipl/cfuncs/mklib.sh || die
+	sed -i -e 's/$(RLINK)/& $(LDFLAGS)/g' src/runtime/Makefile || die
+
+	# executables in src/common are only used to prep the build
+	sed -i -e 's/$(CC) $(CFLAGS)/& $(LDFLAGS)/g' src/common/Makefile || die
+	# executables in src/rtt are only used to perform the build
+	sed -i -e 's/$(CC) $(CFLAGS)/& $(LDFLAGS)/g' src/rtt/Makefile || die
+
+	# do not prestrip files
+	find src -name 'Makefile' | xargs sed -i -e "/strip/d" || die
+}
+
+src_configure() {
+	# select the right compile target.  Note there are many platforms
+	# available
+	local mytarget;
+	if [[ ${CHOST} == *-darwin* ]]; then
+		mytarget="macintosh"
+	else
+		mytarget="linux"
+	fi
+
+	# Fails if more then one make job process.
+	# This is an upstream requirement.
+	emake -j1 $(usex X X-Configure Configure) name=${mytarget}
+
+	# sanitise the Makedefs file generated by Configure
+	sed -i \
+		-e 's:-L/usr/X11R6/lib64::g' \
+		-e 's:-L/usr/X11R6/lib::g' \
+		-e 's:-I/usr/X11R6/include::g' \
+		-e 's:-I/usr/X11R6/include::g' \
+		-e '/^CFLAGS/d' \
+		Makedefs || die "sed of Makedefs failed"
+
+	if use elibc_musl; then
+		append-flags "-D_MUSL"
+	fi
+	append-flags $(test-flags -fno-strict-aliasing -fwrapv)
+}
+
+src_compile() {
+	# Fails if more then one make job process.
+	# This is an upstream requirement.
+	emake -j1 CC="$(tc-getCC)" CFLAGS="${CFLAGS}"
+}
+
+src_test() {
+	emake Samples
+	emake Test
+}
+
+src_install() {
+	# Needed for make Install
+	dodir /usr/$(get_libdir)
+
+	emake Install dest="${D}/usr/$(get_libdir)/icon"
+	dosym ../$(get_libdir)/icon/bin/icont /usr/bin/icont
+	dosym ../$(get_libdir)/icon/bin/iconx /usr/bin/iconx
+	dosym ../$(get_libdir)/icon/bin/icon  /usr/bin/icon
+	if use X; then
+		dosym ../$(get_libdir)/icon/bin/vib /usr/bin/vib
+	fi
+
+	cd "${S}/man/man1" || die
+	doman "${PN}"t.1
+	doman "${PN}".1
+
+	cd "${S}/doc" || die
+	DOCS=( *.txt ../README )
+
+	HTML_DOCS=( *.{htm,gif,jpg,css} )
+	einstalldocs
+
+	# Clean up items from make Install that get installed elsewhere
+	rm -rf "${ED}"/usr/$(get_libdir)/${PN}/man || die
+	rm -rf "${ED}"/usr/$(get_libdir)/icon/{doc,README} || die
+
+	# optional Icon Programming Library
+	if use iplsrc; then
+		cd "${S}" || die
+
+		# Remove unneeded files before copy
+		rm -fv ipl/{BuildBin,BuildExe,CheckAll,Makefile} || die
```

Comment:
Same here, although I'd just always install these unconditionally.

Reply:
Those files are only used at build time and are useless as part of the actual library. I'll make a better comment in the ebuild.